### PR TITLE
Stat updated to return a dict object of splits

### PR DESCRIPTION
**Pull request by Mattsface**
### Why
@KCNilssen wanted stats and splits to be structured in a dict to make it easier to retrieve the wanted stat dated

### What
Updated get stats to return a dict and moved the logic to create stat objects to its own function

New tests were added also

### Tests
pytest

### Risk and impact
- Normal

get_stats has objects as optional now

What is the impact if something does go wrong with this PR?

### CODE REVIEW

File: mlbstatsapi/mlbapi.py [hunk 1]
```diff
@@ -704,10 +704,12 @@ def get_object(self, object):
         # If problem with hydrating object, return the old dry object
         return hydratedobject if hydratedobject else object
 
-    def get_stats(self, object : Union[object, dict], params : dict) -> List[Stats]:
+    def get_stats(self, params : dict, mlb_object : Union[Union[Team, Person], dict] = NoneType) -> List['Splits']:
```
Comment (KCNilssen) [inline mlbstatsapi/mlbapi.py:707]: Should this be `-> Dict{str, 'splits'}`?

Comment (KCNilssen) [reply mlbstatsapi/mlbapi.py:707]: Or something along the lines of that

**Review by KCNilssen — COMMENTED** [review]
LGTM, touch on that comment I left and lets get this good stuff merged!

File: mlbstatsapi/mlbapi.py (hunk 1 above)
Comment (Mattsface) [reply mlbstatsapi/mlbapi.py:707]: hmmm yeah for sure I see

Comment (Mattsface) [reply mlbstatsapi/mlbapi.py:707]: one sec

Comment (Mattsface) [reply mlbstatsapi/mlbapi.py]: Should be all set now @KCNilssen

**Review by KCNilssen — APPROVED** [review]
LGTM

**Pull request merged**